pyuvdata/calfits.py

```diff
@@ -273,8 +285,13 @@ def write_calfits(self, filename, spoof_nonessential=False,
             hdulist = fits.HDUList([prihdu, ant_hdu, sechdu])
 
         else:
-            prihdu = fits.PrimaryHDU(data=pridata, header=prihdr)
-            hdulist = fits.HDUList([prihdu, ant_hdu])
+            if self.total_quality_array is not None:
+                prihdu = fits.PrimaryHDU(data=pridata, header=prihdr)
+                sechdu = fits.ImageHDU(data=secdata, header=sechdr)
+                hdulist = fits.HDUList([prihdu, ant_hdu, sechdu])
+            else:
+                prihdu = fits.PrimaryHDU(data=pridata, header=prihdr)
+                hdulist = fits.HDUList([prihdu, ant_hdu])
 
```

Comment:
It looks to me like you are only handling the total_quality_array if the cal_type is 'gain' and ignoring it if the cal_type is 'delay'. Is this the behavior we want? I think the concept of an overall gain quality should apply equally well for delay type solutions.